### PR TITLE
[dagster-airlift][sensor] top level sensor builder

### PR DESCRIPTION
**Pull request by dpeng817**
## Summary & Motivation
Fix the arguments to the sensor builder to get it ready to be a top level export.
## How I Tested These Changes
Existing tests
## Changelog
NOCHANGELOG

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25054?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25054** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25054?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25053** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25053?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25052** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25052?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25051** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25051?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25050** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25050?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_defs_data.py [hunk 1]
```diff
@@ -24,17 +24,15 @@
 @record
 class AirflowDefinitionsData:
     airflow_instance: AirflowInstance
-    resolved_airflow_defs: Definitions
+    mapped_defs: Definitions
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_defs_data.py:27]: 👍🏻

File: examples/experimental/dagster-airlift/dagster_airlift/core/load_defs.py [hunk 2]
```diff
@@ -68,9 +67,8 @@ def build_defs_from_airflow_instance(
     return Definitions.merge(
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/load_defs.py:67]: <i class='graphite__hidden'>[Re: line +62]<br/><br/></i>

might as well rename all the instances of `resolved_defs` to `mapped_defs` at this point<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25054?utm_source=unchanged-line-comment">Graphite</a>.</span>

Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/core/load_defs.py:67]: Yea I'll do that in a followup I think

File: examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py [hunk 3]
```diff
@@ -65,10 +65,16 @@ def check_keys_for_asset_keys(
 
 
 def build_airflow_polling_sensor_defs(
-    airflow_data: AirflowDefinitionsData,
+    *,
+    mapped_defs: Definitions,
+    airflow_instance: AirflowInstance,
     event_transformer_fn: Optional[DagsterEventTransformerFn],
     minimum_interval_seconds: int = DEFAULT_AIRFLOW_SENSOR_INTERVAL_SECONDS,
 ) -> Definitions:
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py:73]: at this point we should have a docblock explaining this. Not convinced that "mapped" is the right adjective for this top-level API

**Review by schrockn — DISMISSED** [review]
If we are prepping this for being a public API let's write a docblock to make sure it all "fits". Also not convinced re: "mapped" here

File: examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py [hunk 4]
```diff
@@ -77,10 +77,38 @@ def check_keys_for_asset_keys(
...
+    """Builds a sensor that polls Airflow for new dag runs and emits materializations for each asset.
+
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py:87]: The constructed sensor polls the airflow instance for activity inserts materializations (and other events) into Dagster's event log. The sensor decides what dags and tasks to monitor by inspecting the _metadata_ of the passed definitions. This metadata is typically added by `dag_defs` and `task_defs` at definition construction time.

Once the metadata is applied a _mapping_ exists between the airflow entity and the dagster entity. A "mapped" definitions objects this metadata exists on those definitions, and the sensor depends on their existence to function properly.

Comment (schrockn) [reply examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py:87]: ^--- example top-level language we can put somewere

File: examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py [hunk 5]
```diff
@@ -77,10 +77,38 @@ def check_keys_for_asset_keys(
...
+    This mapping is set automatically by `dagster_airlift.core.build_defs_from_airflow_instance` and `dagster_airlift.core.build_mapped_airflow_defs`,
+    but can be set manually as well.
+
+    Using the `event_transformer_fn` argument, users can provide a function that transforms the materializations emitted by the sensor.
+    The expected return type of this function is an iterable of `AssetMaterialization`, `AssetObservation`, or `AssetCheckEvaluation` objects.
+    Each object is expected to have a metadata key `dagster_airlift.constants.EFFECTIVE_TIMESTAMP_METADATA_KEY` which is a `dagster.TimestampMetadataValue` set.
+    This allows Dagster to correctly order the materializations in the event stream.
+
+    Args:
+        mapped_defs (Definitions): The `Definitions` object containing assets mapped to Airflow dags and tasks.
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py:103]: be specific here that this means metadata has been applied.

A "mapped" definitions objects this metadata exists on those definitions, and the sensor depends on their existence to function properly.

**Review by schrockn — APPROVED** [review]
Take another pass at the language, but I've been convinced that "mapped_defs" is a good name for it as long we have a nice definition for it that describes how it means that there is metadata on them the contains mapping information.

**Pull request merged**